common/primitives/src/signatures.rs

```diff
@@ -367,4 +423,90 @@ mod tests {
 		let unified_signer = UnifiedSigner::from(public_key);
 		assert!(!unified_signature.verify(&payload[..], &unified_signer.into_account()));
 	}
+
+	#[test]
+	fn ethereum_address_mapper_should_work_as_expected_for_eth_20_bytes_addresses() {
+		// arrange
+		let eth = from_hex("0x1111111111111111111111111111111111111111").expect("should work");
+
+		// act
+		let account_id = EthereumAddressMapper::to_account_id(&eth);
+		let bytes = EthereumAddressMapper::to_bytes(&eth);
+
+		// assert
+		let expected_address =
+			from_hex("0x1111111111111111111111111111111111111111eeeeeeeeeeeeeeeeeeeeeeee")
+				.expect("should be hex");
+		assert_eq!(account_id, AccountId32::new(expected_address.clone().try_into().unwrap()));
+		assert_eq!(bytes.to_vec(), expected_address);
+	}
+
+	#[test]
+	fn ethereum_address_mapper_should_return_the_same_value_for_32_byte_addresses() {
+		// arrange
+		let eth = from_hex("0x1111111111111111111111111111111111111111111111111111111111111111")
+			.expect("should work");
+
+		// act
+		let account_id = EthereumAddressMapper::to_account_id(&eth);
+		let bytes = EthereumAddressMapper::to_bytes(&eth);
+
+		// assert
+		assert_eq!(account_id, AccountId32::new(eth.clone().try_into().unwrap()));
+		assert_eq!(bytes.to_vec(), eth);
+	}
+
+	#[test]
+	fn ethereum_address_mapper_should_return_the_ethereum_address_with_suffixes_for_64_byte_public_keys(
+	) {
+		// arrange
+		let public_key= from_hex("0x15b5e4aeac2086ee96ab2292ee2720da0b2d3c43b5c699ccdbfd38387e2f71dc167075a80a32fe2c78d7d8780ef1b2095810f12001fa2fcedcd1ffb0aa2ee2c7").expect("should work");
+
+		// act
+		let account_id = EthereumAddressMapper::to_account_id(&public_key);
+		let bytes = EthereumAddressMapper::to_bytes(&public_key);
+
+		// assert
+		let expected_address =
+			from_hex("0x917B536617B0A42B2ABE85AC88788825F29F0B29eeeeeeeeeeeeeeeeeeeeeeee")
+				.expect("should be hex");
```

Comment:
nit: I don't like "magic" generated values, even in tests. Perhaps either a comment how this was derived (ie, `keccak_256 + copy_as_slice(12..)`, or actually derive the value in the test.